pkg/persistence/account/internal/types/types.go

```diff
@@ -16,68 +16,101 @@
 
 package types
 
+import (
+	"fmt"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	ReferenceType          = "reference"
+	PolicyLevelAccount     = "account"
+	PolicyLevelEnvironment = "environment"
+)
+
 type (
-	PolicyLevel = any // either PolicyLevelAccount or PolicyLevelEnvironment is allowed
-	PolicyRef   = any // either string or Reference is allowed
```

Comment:
note: I think we should reflect the change to explicit Level and Reference type(s) for the in-memory structs as well as a followup.

Reply:
in memory types already have a reference type, if you mean that

Reply:
ah nice. Didn't look at them for a while